### PR TITLE
chore: set up section for C4D docs

### PR DESCRIPTION
**Pull request by khorne3**
- [x] Create section for Coder for Docker
- [x] Add template for Docker workspace providers
- [x] Write the docs
- [x] Link Docker workspace provider doc to Coder for Docker docs somehow
- [x] Hide local preview from sidebar and add archive warning

[Notion](https://www.notion.so/coderhq/C4D-Docs-c556ce5c2f2e4b5587eb35574af5f54c)

### CODE REVIEW

**Comment by Emyrk** [comment]
TODO:  `-e PROXY_TRUSTED_ORIGINS=172.17.0.0/16` and `-e PROXY_TRUSTED_HEADERS=X-Forwarded-For`

**Comment by Emyrk** [comment]
Working on this today

File: setup/coder-for-docker/postgres.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+title: "External database setup"
+description: Learn how to set up an external Postgres database for use with C4D.
+---
```
Comment (Emyrk) [inline setup/coder-for-docker/postgres.md:4]: Should we move the docs from here?
https://coder.com/docs/coder/latest/setup/docker#use-an-external-postgresql-database

File: setup/coder-for-docker/cloud.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+title: "Cloud deployment"
+description: Learn how to run Coder with Docker in the cloud.
+---
+
+
+
+Forthcoming.
```
Comment (Emyrk) [inline setup/coder-for-docker/cloud.md:8]: What should I put here?

**Review by Emyrk — COMMENTED** [review]
I added the docs to setup remote C4D providers

File: setup/coder-for-docker/cloud.md (hunk 2 above)
Comment (khorne3) [reply setup/coder-for-docker/cloud.md:8]: I think this is where @bpmct's stuff will live, but we might hold off on releasing this.

File: setup/coder-for-docker/index.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+title: "Coder for Docker"
+description: Learn how to run Coder with Docker.
+---
+
+<children></children>
```
[the remark this answers is not in the document]
Comment (khorne3) [reply setup/coder-for-docker/index.md:6]: Nothing needed; it will automatically populate little squares that link to the docs in this section.

**Review by Emyrk — APPROVED** [review]

**Pull request merged**